internal/infrastructure/oceanwallet/notifications_manager.go

```diff
@@ -0,0 +1,67 @@
+package oceanwallet
+
+import (
+	"context"
+
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+	oceanv1alpha "github.com/vulpemventures/ocean/api-spec/protobuf/gen/go/ocean/v1alpha"
+	"google.golang.org/grpc"
+)
+
+type notificationsManagerGrpc struct {
+	client oceanv1alpha.NotificationServiceClient
+}
+
+var _ ports.NotificationManager = (*notificationsManagerGrpc)(nil)
+
+func newNotificationsManagerGrpc(conn *grpc.ClientConn) ports.NotificationManager {
+	return &notificationsManagerGrpc{
+		client: oceanv1alpha.NewNotificationServiceClient(conn),
+	}
+}
+
+func (nm *notificationsManagerGrpc) TxChannel() (chan ports.TxNotification, error) {
+	req := &oceanv1alpha.TransactionNotificationsRequest{}
+	stream, err := nm.client.TransactionNotifications(context.Background(), req)
+	if err != nil {
+		return nil, err
+	}
+
+	txChan := make(chan ports.TxNotification)
+
+	go func() {
+		for {
+			resp, err := stream.Recv()
+			if err != nil {
+				return
+			}
+
+			txChan <- &txNotificationGrpc{resp}
+		}
+	}()
+
+	return txChan, nil
+}
+
+func (nm *notificationsManagerGrpc) UtxoChannel() (chan ports.UtxoNotification, error) {
+	req := &oceanv1alpha.UtxosNotificationsRequest{}
+	stream, err := nm.client.UtxosNotifications(context.Background(), req)
+	if err != nil {
+		return nil, err
+	}
+
+	utxoChan := make(chan ports.UtxoNotification)
+
+	go func() {
+		for {
+			resp, err := stream.Recv()
+			if err != nil {
+				return
+			}
+
+			utxoChan <- &utxoNotificationGrpc{resp}
```

Comment:
so, ideally here is where we might need to enrich the notification with block details etc for spent utxos, right?

Reply:
I don't think so,we need to access the `transactionManager` in order to call `GetTransaction` rpc. I think this should be done in the upper layer (application) **only when the blockdetails and txdetails are needed** : see `tradeSettleOrExpire` function as an exemple, what's your opinion ?

Reply:
ok fine, at app level we will eventually call the `TransactionManager().GetTransactionById(...)`  API to retrieve missing data if needed.